velox/type/Timestamp.cpp

```diff
@@ -247,6 +247,7 @@ std::string Timestamp::tmToString(
   if (options.mode != TimestampToStringOptions::Mode::kTimeOnly) {
     int n = kTmYearBase + tmValue.tm_year;
     bool negative = n < 0;
+    const bool leadingPositiveSign = options.leadingPositiveSign && n > 9999;
```

Comment:
It is here because the code after changes the value of `n`.
https://github.com/facebookincubator/velox/pull/8165/files#diff-6dea200bd5a075037648456dfd3eb589c84b009eb4c7f11d559f23ac7cba1cd5R251-R258